src/main/java/com/artipie/scripting/Script.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+package com.artipie.scripting;
+
+import com.artipie.ArtipieException;
+import java.util.Map;
+import java.util.Objects;
+import javax.script.Compilable;
+import javax.script.CompiledScript;
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+import javax.script.SimpleBindings;
+import javax.script.SimpleScriptContext;
+
+/**
+ * Script.
+ *
+ * @since 0.30
+ */
+public interface Script {
+    /**
+     * Script engine manager.
+     */
+    ScriptEngineManager MANAGER = new ScriptEngineManager();
+
+    /**
+     * Call script.
+     * @return Result of script execution.
+     * @throws ScriptException Script execution exception
+     */
+    Result call() throws ScriptException;
+
+    /**
+     * Call script by passing environment variables.
+     * @param vars Environment variables
+     * @return Result of script execution.
+     * @throws ScriptException Script execution exception
+     */
+    Result call(Map<String, Object> vars) throws ScriptException;
+
+    /**
+     * Create instance of {@link Script}.
+     * @param name Name of scripting engine.
+     * @param script Script code
+     * @return Script
+     */
+    @SuppressWarnings("PMD.ProhibitPublicStaticMethods")
+    static Script newScript(final String name, final String script) {
+        final ScriptEngine engine = Script.MANAGER.getEngineByName(name);
+        return new StandardScript(engine, script);
+    }
+
+    /**
+     * Create precompiled instance of {@link Script}.
+     * @param name Name of scripting engine.
+     * @param script Script code
+     * @return Script
+     */
+    @SuppressWarnings("PMD.ProhibitPublicStaticMethods")
+    static Script newCompiledScript(final String name, final String script) {
+        final ScriptEngine engine = Script.MANAGER.getEngineByName(name);
+        return new PrecompiledScript(engine, script);
+    }
```

Comment:
Why do we need these static methods? I'd simply added secondary constructors, which accepts string as engine name and call `Script.MANAGER.getEngineByName(name)` in the call to primary ctor.